app/services/the_tvdb_service.rb

```diff
@@ -1,41 +1,58 @@
 class TheTvdbService
+  class NotFound < StandardError; end
+
   BASE_URL = 'https://api.thetvdb.com'.freeze
   API_KEY = ENV['THE_TVDB_API_KEY'].freeze
 
-  # This will create new mappings that combine series/season together.
-  def create_tvdb_mapping!
-    return if mapping_data.blank?
+  def initialize(set_name)
+    @set_name = set_name
+  end
 
-    mapping_data['Anime'].each do |media_ids|
-      # set the media_id (this is for our database)
-      media_id = media_ids[0]
-      series_id = media_ids[1]['thetvdb/series']
-      season_id = media_ids[1]['thetvdb/season']
+  def import!
+    data = query_data
+    return if data.blank?
 
-      response = get(build_episode_path(series_id, season_id))
+    data['Anime'].each do |(anime_id, mappings)|
+      series_id, season_id = mappings['thetvdb'].split('/')
 
-      next if response.code == 404
-      raise 'TVDB Error' unless response.success?
+      response = get(build_episode_path(series_id, season_id))
+      next unless response
 
-      response = JSON.parse(response.body)
+      # update the episode count if it does not exist.
+      tvdb_episodes_count = response['data'].count
+      anime = update_anime_episode_count(anime_id, tvdb_episodes_count)
```

Comment:
`update_anime_episode_count` is doing more than the name implies, it's also acting as `Anime.find`, which makes this really odd to read.  Why would we call `update_anime_episode_count` with an `anime_id` and `episodes_count` and then... store the result as `anime`????

It's probably best to just do that right here instead of in a method